planner/cascades/testdata/transformation_rules_suite_out.json

```diff
@@ -219,14 +219,108 @@
+          "    TopN_11 input:[Group#3], test.t.a:asc, offset:0, count:1",
+          "Group#3 Schema:[test.t.a,test.t.c]",
+          "    TiKVSingleGather_7 input:[Group#4], table:t",
+          "    TiKVSingleGather_9 input:[Group#5], table:t, index:c_d_e",
+          "Group#4 Schema:[test.t.a,test.t.c]",
+          "    TableScan_6 table:t, pk col:test.t.a",
+          "Group#5 Schema:[test.t.a,test.t.c]",
+          "    IndexScan_8 table:t, index:c, d, e"
+        ]
+      },
+      {
+        "SQL": "select c from t order by t.a + t.b limit 1",
+        "Result": [
+          "Group#0 Schema:[test.t.c]",
+          "    Projection_5 input:[Group#1], test.t.c",
+          "Group#1 Schema:[test.t.c,test.t.a,test.t.b]",
+          "    Projection_2 input:[Group#2], test.t.c, test.t.a, test.t.b",
+          "Group#2 Schema:[test.t.a,test.t.b,test.t.c]",
+          "    TopN_9 input:[Group#3], plus(test.t.a, test.t.b):asc, offset:0, count:1",
+          "Group#3 Schema:[test.t.a,test.t.b,test.t.c]",
+          "    TiKVSingleGather_7 input:[Group#4], table:t",
+          "Group#4 Schema:[test.t.a,test.t.b,test.t.c]",
+          "    TableScan_6 table:t, pk col:test.t.a"
+        ]
+      },
+      {
+        "SQL": "select 1 as a from dual order by a limit 1",
+        "Result": [
+          "Group#0 Schema:[Column#1]",
+          "    Projection_2 input:[Group#1], 1->Column#1",
+          "Group#1 Schema:[]",
+          "    TopN_6 input:[Group#2], 1:asc, offset:0, count:1",
+          "Group#2 Schema:[]",
+          "    TableDual_1 rowcount:1"
+        ]
+      },
+      {
+        "SQL": "select * from t t1 where t1.a in (select t2.a as a from t t2 where t2.b > t1.b order by t1.b limit 1)",
+        "Result": [
+          "Group#0 Schema:[test.t.a,test.t.b,test.t.c,test.t.d,test.t.e,test.t.c_str,test.t.d_str,test.t.e_str,test.t.f,test.t.g,test.t.h,test.t.i_date]",
+          "    Projection_9 input:[Group#1], test.t.a, test.t.b, test.t.c, test.t.d, test.t.e, test.t.c_str, test.t.d_str, test.t.e_str, test.t.f, test.t.g, test.t.h, test.t.i_date",
+          "Group#1 Schema:[test.t.a,test.t.b,test.t.c,test.t.d,test.t.e,test.t.c_str,test.t.d_str,test.t.e_str,test.t.f,test.t.g,test.t.h,test.t.i_date]",
+          "    Apply_8 input:[Group#2,Group#3], semi join, equal:[eq(test.t.a, test.t.a)]",
+          "Group#2 Schema:[test.t.a,test.t.b,test.t.c,test.t.d,test.t.e,test.t.c_str,test.t.d_str,test.t.e_str,test.t.f,test.t.g,test.t.h,test.t.i_date]",
+          "    TiKVSingleGather_11 input:[Group#4], table:t1",
+          "Group#4 Schema:[test.t.a,test.t.b,test.t.c,test.t.d,test.t.e,test.t.c_str,test.t.d_str,test.t.e_str,test.t.f,test.t.g,test.t.h,test.t.i_date]",
+          "    TableScan_10 table:t1, pk col:test.t.a",
+          "Group#3 Schema:[test.t.a]",
+          "    Projection_5 input:[Group#5], test.t.a",
+          "Group#5 Schema:[test.t.a,test.t.b]",
+          "    TopN_15 input:[Group#6], , offset:0, count:1",
```

Comment:
If the TopN has no columns, we can just remove it?

Reply:
I am not sure how could we determine if the TopN has no columns in this case?

Reply:
After removing meaningless constant sort items, you can check that?

Reply:
Should it like this after removed?
```
{
        "SQL": "select 1 as a from dual limit 1",
        "Result": [
          "Group#0 Schema:[Column#1]",
          "    Limit_3 input:[Group#1], offset:0, count:1",
          "Group#1 Schema:[Column#1]",
          "    Projection_2 input:[Group#2], 1->Column#1",
          "Group#2 Schema:[]",
          "    TableDual_1 rowcount:1"
        ]
      },
```

Reply:
You should convert this `select a, b from (select 1 as a, b from t)t2 order by a limit 10` to `select a, b from (select 1 as a, b from t)t2 limit 10`. You can see that the `order by a` actually orders by a constant.

So after you remove those meaningless constant items, you should check if this topN's ByItems is empty. If so, transform this TopN to a Limit.